packages/expo-modules-core/ios/ReactDelegates/ExpoReactDelegate.swift

```diff
@@ -21,10 +21,12 @@ public class ExpoReactDelegate: NSObject {
       .compactMap { $0.createReactRootView(reactDelegate: self, moduleName: moduleName, initialProperties: initialProperties, launchOptions: launchOptions) }
       .first(where: { _ in true })
       ?? {
-        guard let rctAppDelegate = (UIApplication.shared.delegate as? RCTAppDelegate) else {
-          fatalError("The `UIApplication.shared.delegate` is not a `RCTAppDelegate` instance.")
+        guard let appDelegate = (UIApplication.shared.delegate as? ExpoReactNativeFactoryDelegate) ??
+          ((UIApplication.shared.delegate as? NSObject)?.value(forKey: "_expoAppDelegate") as? ExpoReactNativeFactoryDelegate) else {
+          fatalError("`UIApplication.shared.delegate` must be an `ExpoAppDelegate` or `EXAppDelegateWrapper`")
```

Comment:
in the future we should have a better way to deal with this, for RCTAppDelegate dreprecation.

Reply:
Absolutely - I think this was more a "get-it-to-compile" fix - and maybe not meant for production. The issue is of course that we have multiple paths to get to the `ExpoReactNativeFactoryDelegate` unfortunately.

Reply:
Any ideas would be welcome!

Reply:
@Kudo any ETA for this to be fixed?

Reply:
@mdjastrzebski does it block anything from you?